src/WithdrawRequestNFT.sol

```diff
@@ -120,13 +126,33 @@ contract WithdrawRequestNFT is ERC721Upgradeable, UUPSUpgradeable, OwnableUpgrad
         }
     }
 
-    // a function to transfer accumulated shares to admin
-    function burnAccumulatedDustEEthShares() external onlyAdmin {
-        require(eETH.totalShares() > accumulatedDustEEthShares, "Inappropriate burn");
-        uint256 amount = accumulatedDustEEthShares;
-        accumulatedDustEEthShares = 0;
+    // There have been errors tracking `accumulatedDustEEthShares` in the past.
+    // - https://github.com/etherfi-protocol/smart-contracts/issues/24
+    // This is a one-time function to handle the remainder of the eEth shares after the claim of the withdraw requests
+    // It must be called only once with ALL the requests that have not been claimed yet.
+    // there are <3000 such requests and the total gas spending is expected to be ~9.0 M gas.
+    function handleAccumulatedShareRemainder(uint256[] memory _reqIds) external onlyOwner {        
```

Comment:
I think this method will likely fail given that it must be called via timelock:

- timelock transaction is scheduled with valid `_reqIds` array
- users claim withdrawals from the `_reqIds` array during 3 day timelock enforced delay
- timelock transaction then reverts when we go to execute it

See POC of failure:
https://github.com/etherfi-protocol/smart-contracts/pull/214/files

Reply:
resolved!